codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -169,7 +172,8 @@ private static ByteBuf encodeConnectMessage(
 
                 int variablePartSize = variableHeaderBufferSize + payloadBufferSize;
                 int fixedHeaderBufferSize = 1 + getVariableLengthInt(variablePartSize);
-                ByteBuf buf = ctx.alloc().buffer(fixedHeaderBufferSize + variablePartSize);
+                int totalSize = fixedHeaderBufferSize + variablePartSize;
+                ByteBuf buf = ctx.alloc().buffer(totalSize, totalSize);
```

Comment:
I don't think these changes to set max capacity == init capacity should be necessary, it won't affect how the buffer is allocated just the cap on how large it can grow beyond init capacity (which it won't because it's being sized precisely up-front IIUC).